### PR TITLE
 [TechDraw] Improve readability of DrawViewPart.cpp replaces #7208

### PR DESCRIPTION
**Pull request by WandererFan**
[TechDraw] Improve readability of DrawViewPartPyImp.cpp

Thank you for creating a pull request to contribute to FreeCAD! To ease integration, we ask you to conform to the following items. Pull requests which don't satisfy all the items below might be rejected. If you are in doubt with any of the items below, don't hesitate to ask for help in the [FreeCAD forum](https://forum.freecadweb.org/viewforum.php?f=10)!

- [ ]  Your pull request is confined strictly to a single module. That is, all the files changed by your pull request are either in `App`, `Base`, `Gui` or one of the `Mod` subfolders. If you need to make changes in several locations, make several pull requests and wait for the first one to be merged before submitting the next ones
- [ ]  In case your pull request does more than just fixing small bugs, make sure you discussed your ideas with other developers on the FreeCAD forum
- [ ]  Your branch is [rebased](https://git-scm.com/docs/git-rebase) on latest master `git pull --rebase upstream master`
- [ ]  All FreeCAD unit tests are confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [ ]  All commit messages are [well-written](https://chris.beams.io/posts/git-commit/) ex: `Fixes typo in Draft Move command text`
- [ ]  Your pull request is well written and has a good description, and its title starts with the module name, ex: `Draft: Fixed typos`
- [ ]  Commit messages include `issue #<id>` or `fixes #<id>` where `<id>` is the issue ID number from our [Issues database](https://github.com/FreeCAD/FreeCAD/issues) in case a particular commit solves or is related to an existing issue. Ex: `Draft: fix typos - fixes #4805`

And please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [0.20 Changelog Forum Thread](https://forum.freecadweb.org/viewtopic.php?f=10&t=56135).

---

### CODE REVIEW

**Comment by benj5378** [comment]
https://github.com/FreeCAD/FreeCAD/blob/158444701fac93abd3092c4c654384e6c79bc523/src/Mod/TechDraw/App/DrawViewPart.cpp#L627-L628
```
        if ((*it)->getTypeId().isDerivedFrom(DrawHatch::getClassTypeId()) &&
            !(*it)->isRemoving()) {
```

https://github.com/FreeCAD/FreeCAD/blob/158444701fac93abd3092c4c654384e6c79bc523/src/Mod/TechDraw/App/DrawViewPart.cpp#L641
```
        if ((*it)->getTypeId().isDerivedFrom(DrawGeomHatch::getClassTypeId()) &&
            !(*it)->isRemoving()) {
```

https://github.com/FreeCAD/FreeCAD/blob/158444701fac93abd3092c4c654384e6c79bc523/src/Mod/TechDraw/App/DrawViewPart.cpp#L803
```
    return bbx.MaxX - bbx.MinX;
```
https://github.com/FreeCAD/FreeCAD/blob/158444701fac93abd3092c4c654384e6c79bc523/src/Mod/TechDraw/App/DrawViewPart.cpp#L809
```
    return bbx.MaxX - bbx.MinX;
```
https://github.com/FreeCAD/FreeCAD/blob/158444701fac93abd3092c4c654384e6c79bc523/src/Mod/TechDraw/App/DrawViewPart.cpp#L935-L939
```
        if (o->getTypeId().isDerivedFrom(DrawViewDetail::getClassTypeId()) &&
            !o->isRemoving()) {
            result.push_back(static_cast<TechDraw::DrawViewDetail*>(o));
        }
```
https://github.com/FreeCAD/FreeCAD/blob/158444701fac93abd3092c4c654384e6c79bc523/src/Mod/TechDraw/App/DrawViewPart.cpp#L1106-L1108
```
    m_referenceVerts.push_back(ref);
    return ref->getTagAsString();
```

File: src/Mod/TechDraw/App/DrawViewPart.cpp [hunk 1]
```diff
@@ -392,7 +381,7 @@ void DrawViewPart::addShapes2d(void)
             Base::Vector3d projected = projectPoint(vp * getScale());
             TechDraw::VertexPtr v1(std::make_shared<TechDraw::Vertex>(projected));
             geometryObject->addVertex(v1);
-        } else if (s.ShapeType() == TopAbs_EDGE) {
+        } // else if (s.ShapeType() == TopAbs_EDGE) {
```
Comment (AjinkyaDahale) [inline src/Mod/TechDraw/App/DrawViewPart.cpp:384]: Why not just remove it altogether?

File: src/Mod/TechDraw/App/DrawViewPart.cpp [hunk 2]
```diff
@@ -773,16 +754,18 @@ std::vector<TechDraw::BaseGeomPtr> DrawViewPart::getFaceEdgesByIndex(int idx) co
...
-        TechDraw::FacePtr projFace = faces.at(idx);
-        for (auto& w: projFace->wires) {
-            for (auto& g:w->geoms) {
-                if (g->cosmetic) {
-                    //if g is cosmetic, we should skip it
-                    Base::Console().Log("DVP::getFaceEdgesByIndex - found cosmetic edge\n");
-                } else {
-                    result.push_back(g);
-                }
+    if (idx >= (int) faces.size()) {
```
Comment (AjinkyaDahale) [inline src/Mod/TechDraw/App/DrawViewPart.cpp:757]: C-style cast here.

**Comment by berndhahnebach** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7234">pipeline status</a> for feature branch [PR_7234](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7234). Pipeline [600445249](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/600445249) was triggered at [cf468f7](https://github.com/FreeCAD/FreeCAD/pull/7234/commits/cf468f73778c29d9eee5fd448ae27cedaf8b54fe). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by WandererFan** [comment]
closed as stale branch with merge conflicts.

**Pull request closed without merging**